pages/index.js

```diff
@@ -14,8 +16,26 @@ export default function HomePage() {
   if (error) {
     return <h1>Something went wrong!</h1>;
   }
+
+  if (!pieces) {
+    return <p>Keine Daten vorhanden</p>;
+  }
+
+  function randomIndex() {
```

Comment:
This is a good Solution - however - you could write this in one line:

  const randomPiece = pieces[Math.floor(Math.random() * pieces.length)];

:)